pandas/core/frame.py

```diff
@@ -11259,7 +11286,43 @@ def quantile(
             res = self._constructor([], index=q, columns=cols, dtype=dtype)
             return res.__finalize__(self, method="quantile")
 
-        res = data._mgr.quantile(qs=q, axis=1, interpolation=interpolation)
+        valid_method = {"single", "table"}
```

Comment:
maybe i am not getting something, but why isn't this just
```np.asarray(res_df).ravel()``` and then reed to the existing quantile routine?

Reply:
1. I think that approach would not work for DataFrames with mixed dtypes
2. For the limited set of interpolation methods supported (to start) in this PR, I think this approach is more performant as only quantile indices are calculated followed by a `take`.